GitFyle.Core.Api.Tests.Unit/Services/Foundations/Contributions/ContributionServiceTests.cs

```diff
@@ -0,0 +1,89 @@
+using GitFyle.Core.Api.Services.Foundations.Contributions;
+using Microsoft.Data.SqlClient;
+using Moq;
+using Tynamix.ObjectFiller;
+using Xeptions;
+
+namespace GitFyle.Core.Api.Tests.Unit.Services.Foundations.Contributions
+{
+    public partial class ContributionServiceTests
+    {
+        private readonly Mock<IStorageBroker> storageBrokerMock;
+        private readonly Mock<IDateTimeBroker> dateTimeBrokerMock;
+        private readonly Mock<ILoggingBroker> loggingBrokerMock;
+        private readonly ContributionService contributionService;
+
+        public ContributionServiceTests()
+        {
+            this.storageBrokerMock = new Mock<IStorageBroker>();
+            this.dateTimeBrokerMock = new Mock<IDateTimeBroker>();
+            this.loggingBrokerMock = new Mock<ILoggingBroker>();
+
+            this.contributionService = new ContributionService(
+                storageBroker: this.storageBrokerMock.Object,
+                dateTimeBroker: this.dateTimeBrokerMock.Object,
+                loggingBroker: this.loggingBrokerMock.Object);
+        }
+
+        private static Expression<Func<Xeption, bool>> SameExceptionAs(
+            Xeption expectedException)
+        {
+            return actualException =>
+                actualException.SameExceptionAs(expectedException);
+        }
+
+        private SqlException CreateSqlException()
+        {
+            return (SqlException)RuntimeHelpers.GetUninitializedObject(
+                type: typeof(SqlException));
+        }
+
+        private static DateTimeOffset GetRandomDateTimeOffset() =>
+            new DateTimeRange(earliestDate: new DateTime()).GetValue();
+
+        private static string GetRandomStringWithLengthOf(int length)
+        {
+            return new MnemonicString(wordCount: 1, wordMinLength: length, wordMaxLength: length)
+                .GetValue();
+        }
+
+        private static string GetRandomString() =>
+            new MnemonicString().GetValue();
+
+        private static Contribution CreateRandomContribution() =>
+            CreateRandomContribution(dateTimeOffset: GetRandomDateTimeOffset());
+
+        private static Contribution CreateRandomContribution(DateTimeOffset dateTimeOffset) =>
+            CreateContributionFiller(dateTimeOffset).Create();
+
+        private static Filler<Contribution> CreateContributionFiller(DateTimeOffset dateTimeOffset)
+        {
+            string user = Guid.NewGuid().ToString();
+            var filler = new Filler<Contribution>();
+
+            filler.Setup()
+                .OnType<DateTimeOffset>().Use(dateTimeOffset)
+                .OnProperty(contribution => contribution.ExternalCreatedAt).Use(dateTimeOffset)
+                .OnProperty(contribution => contribution.ExternalUpdatedAt).Use(dateTimeOffset)
+                .OnProperty(contribution => contribution.ExternalMergedAt).Use(dateTimeOffset)
+                .OnProperty(contribution => contribution.CreatedBy).Use(user)
```

Comment:
We can remove these 3 as they are covered by the `.OnType<DateTimeOffset>().Use(dateTimeOffset)`

                .OnProperty(contribution => contribution.ExternalCreatedAt).Use(dateTimeOffset)
                .OnProperty(contribution => contribution.ExternalUpdatedAt).Use(dateTimeOffset)
                .OnProperty(contribution => contribution.ExternalMergedAt).Use(dateTimeOffset)

```suggestion
            filler.Setup()
                .OnType<DateTimeOffset>().Use(dateTimeOffset)
                .OnProperty(contribution => contribution.CreatedBy).Use(user)
```

Reply:
@SlimAhmad fix this and we can merge your work.